datasets/_analysis/ann_graph_topics/analyzer.py

```diff
@@ -6,57 +8,72 @@
+    topic: str,
+    existing_topics: Set[str],
+) -> None:
+    context.log.info(
+        f"Adding topic: {topic}",
+        source=risk_source.caption,
+        source_id=risk_source.id,
+        linked=related_entity.caption,
+        linked_id=related_entity.id,
+        topics=existing_topics,
+    )
+
+    patch = context.make(related_entity.schema)
+    patch.id = related_entity.id
+    patch.add("topics", topic)
+    context.emit(patch)
+
+
 def analyze_entity(context: Context, view: View, entity: Entity) -> None:
     topics = entity.get_type_values(registry.topic)
     for prop, adjacent in view.get_adjacent(entity):
         asch = adjacent.schema
-        if prop.reverse is None or not asch.edge:
-            continue
 
+        # For when the other entity is on the other side of an edge
         other_prop = (
             asch.source_prop if prop.reverse == asch.target_prop else asch.target_prop
         )
-        if other_prop is None:
-            continue
 
+        # Tag role.rca for family relations of PEPs
         if "role.pep" in topics and adjacent.schema.is_a("Family"):
             for other_id in adjacent.get(other_prop):
                 other = view.get_entity(other_id)
                 if other is None or not other.schema.is_a("Person"):
                     continue
-                other_topic_stmts = other.get_statements("topics")
-                other_topics = [
-                    s.value
-                    for s in other_topic_stmts
-                    if s.dataset != context.dataset.name
-                ]
-                if "role.rca" in other_topics or "role.pep" in other_topics:
+                other_topics = non_graph_topics(context, other)
+                if other_topics.intersection({"role.rca", "role.pep"}):
                     continue
-                context.log.info(
-                    "Adding topic: role.rca",
-                    pep=entity.caption,
-                    pep_id=entity.id,
-                    rca=other.caption,
-                    rca_id=other.id,
-                    topics=other_topics,
-                )
-                patch = context.make(other.schema)
-                patch.id = other.id
-                patch.add("topics", "role.rca")
-                context.emit(patch)
+                emit_patch(context, entity, other, "role.rca", other_topics)
 
         # Family is eternal, business is time-bound:
         if len(adjacent.get("endDate", quiet=True)) > 0:
             context.log.info(
-                "Skipping entity with end date",
-                adjacent=adjacent.id,
-                entity=entity.id,
+                "Skipping entity with end date", adjacent=adjacent.id, entity=entity.id
             )
             continue
 
+        # Tag sanction.linked for sanction-linked entities
         if "sanction" in topics:
             if entity.schema.is_a("Security"):
                 continue
+            if adjacent.schema.is_a("Security"):
```

Comment:
Are there any other links than `issuer` that we want to get here? Maybe worth a comment what the intention here is, for the next time I'm trying to figure out what's actually going on here :)